pkg/openstack/loadbalancer_test.go

```diff
@@ -534,3 +536,104 @@ func Test_getListenerProtocol(t *testing.T) {
 		})
 	}
 }
+
+func TestLbaasV2_getMemberSubnetID(t *testing.T) {
+	lbaas := &LbaasV2{
+		LoadBalancer: LoadBalancer{
+			opts: LoadBalancerOpts{
+				LBClasses:      map[string]*LBClass{"lbclassKey": {MemberSubnetID: "lb-class-member-subnet-id-5678"}},
+				MemberSubnetID: "lb-class-memberSubnetId",
+			},
+		},
+	}
+
+	type fields struct {
+		lbaas LbaasV2
+	}
+	type args struct {
+		service *corev1.Service
+		svcConf *serviceConfig
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    string
+		wantErr error
+	}{
+		{
+			name:   "get member subnet id from service annotation",
+			fields: fields{},
+			args: args{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							ServiceAnnotationLoadBalancerMemberSubnetID: "member-subnet-id",
+						},
+					},
+				},
+				svcConf: &serviceConfig{},
+			},
+			want:    "member-subnet-id",
+			wantErr: nil,
+		},
+		{
+			name: "get member subnet id from config class",
+			fields: fields{
+				lbaas: *lbaas,
+			},
+			args: args{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							ServiceAnnotationLoadBalancerClass: "lbclassKey",
+						},
+					},
+				},
+				svcConf: &serviceConfig{},
+			},
+			want:    "lb-class-member-subnet-id-5678",
+			wantErr: nil,
+		},
+		{
+			name:   "get member subnet id from default config",
+			fields: fields{},
+			args: args{
+				service: &corev1.Service{},
+				svcConf: &serviceConfig{},
+			},
+			want:    "",
+			wantErr: nil,
+		},
+		{
+			name:   "error when loadbalancer class not found",
+			fields: fields{},
+			args: args{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							ServiceAnnotationLoadBalancerClass: "invalid-lb-class",
+						},
+					},
+				},
+				svcConf: &serviceConfig{},
+			},
+			want:    "",
+			wantErr: fmt.Errorf("invalid loadbalancer class \"invalid-lb-class\""),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			lbaas := tt.fields.lbaas
+
+			got, err := lbaas.getMemberSubnetID(tt.args.service, tt.args.svcConf)
+			if (err != nil) != (tt.wantErr != nil) || (err != nil && err.Error() != tt.wantErr.Error()) {
+				t.Errorf("LbaasV2.getMemberSubnetID() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
```

Comment:
You should rather do something like this:

```
if tt.wantErr != nil {
    assert.ErrorIs(...)
} else {
    assert.NoError(...)
}
```
Maybe even `assert.ErrorIs` can be used with a `nil` target to make it one line?